### ISSUE TITLE
Publish-PerTenantExtensionApps: (503) Server Unavailable

### ISSUE THREAD

**Issue by vvkjndl** [microsoft/navcontainerhelper#2371]
**Describe the issue**
We are using the function `Publish-PerTenantExtensionApps` to deploy applications on BC Cloud tenant using `bcauthcontext`. At the end of installation the function throws an error (503) Server Unavailable.

However, BC Cloud tenant reports that the deployment of app was successful (please refer to the screenshot).

**Scripts used to create container and cause the issue**
```
Publish-PerTenantExtensionApps `
    -bcAuthContext $bcauthcontext `
    -environment "$(Environment)" `
    -appFiles @($($appFiles.FullName)) `
    -useNewLine `
    -ErrorAction Continue
```

**Full output of scripts**
```
2022-03-23T14:58:56.8665431Z iNECTA LLC_YIBASE - Base App_2020.10.12.003.app
2022-03-23T14:58:59.7970287Z Publishing and Installing
2022-03-23T14:59:11.5560684Z .
2022-03-23T14:59:17.3924829Z .
2022-03-23T14:59:22.7324147Z .
2022-03-23T14:59:28.0408337Z .
2022-03-23T14:59:33.3073577Z .
2022-03-23T14:59:38.5999646Z .
2022-03-23T14:59:43.9857064Z .
2022-03-23T14:59:49.3345286Z .
2022-03-23T14:59:54.6668713Z .
2022-03-23T14:59:59.9890632Z .
2022-03-23T15:00:05.2816064Z .
2022-03-23T15:00:30.5238216Z The remote server returned an error: (503) Server Unavailable.
2022-03-23T15:00:31.2021173Z Publish-PerTenantExtensionApps Telemetry Correlation Id: a82073bd-1458-48c3-accb-0676f9a6ffe4
2022-03-23T15:00:31.4284786Z ##[error]Invoke-WebRequest : The remote server returned an error: (503) Server Unavailable.
At C:\Program Files\WindowsPowerShell\Modules\BcContainerHelper\3.0.3\BcSaaS\Publish-PerTenantExtensionApps.ps1:146 
char:30
+ ... xtensions = Invoke-WebRequest -Headers $authHeaders -Method Get -Uri  ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (System.Net.HttpWebRequest:HttpWebRequest) [Invoke-WebRequest], WebExc 
   eption
    + FullyQualifiedErrorId : WebCmdletWebResponseException,Microsoft.PowerShell.Commands.InvokeWebRequestCommand
2022-03-23T15:00:31.4333493Z ##[error]PowerShell exited with code '1'.
```

**Screenshots**
![pte-error](https://user-images.githubusercontent.com/1667969/159896986-3c396771-1231-4d8b-934b-ee8dd56401b4.png)

**Additional context**
- does it happen all the time? Yes
- did it use to work? Yes

**Comment by freddydk** [comment]
This might be a bug, will investigate

**Comment by vvkjndl** [comment]
@freddydk Increasing the sleep time from 10 seconds to 30 seconds resolved the issue from me.

Shall we increase the sleep time to 30 seconds in upstream master, what do you say?

**Comment by freddydk** [comment]
We need to find a more resilient way - will investigate

**Comment by freddydk** [comment]
like - retry x times for some errors

**Comment by freddydk** [comment]
Thanks for finding out

**Comment by freddydk** [comment]
Shipped in 3.0.5
